numpy/core/shape_base.py

```diff
@@ -177,6 +169,87 @@ def atleast_3d(*arys):
         return res
 
 
+def atleast_nd(ary, n, pos='start'):
+    """
+    View input as array with at least `n` dimensions.
+
+    Parameters
+    ----------
+    ary : array_like
+        The input array. Non-array inputs are converted to arrays.
+        Arrays that already have `n` or more dimensions are preserved.
+    n : scalar
+        The minimum number of dimensions required.
+    pos : {'start', 'end'}, optional
+        Where to add new dimensions. All new dimensions are added on the
+        same side of the shape. Default is 'start'.
+
+    Returns
+    -------
+    res : ndarray
+        An array with ``res.ndim >= n``. A view is returned for array
+        inputs. Dimensions are prepended if `pos` is 'start', so for
+        example, a 1-D array of shape ``(N,)`` with ``n=4`` becomes a
+        view of shape ``(1, 1, 1, N)``. Dimensions are appended if
+        `pos` is 'end', so for example a 2-D array of shape ``(M, N)``
+        becomes a view of shape ``(M, N, 1, 1)`` when ``n=4``.
+
+    See Also
+    --------
+    atleast_1d, atleast_2d, atleast_3d
+
+    Notes
+    -----
+    This function does not follow the convention of the other atleast_*d
+    functions in numpy in that it only accepts a single array argument.
+    To process multiple arrays, use a comprehension or loop around the
+    function call. See examples below.
+
+    Setting ``pos='start'`` is equivalent to how the array would be
+    interpreted by numpy's broadcasting rules. There is no need to call
+    this function for simple broadcasting.
+
+    Setting ``pos='end'`` is roughly (but not exactly) equivalent to
+    doing ``np.array(ary, copy=False, subok=True, ndmin=n)``.
+
+    Examples
+    --------
+    >>> np.atleast_nd(3.0, 4)
+    array([[[[ 3.]]]])
+
+    >>> x = np.arange(3.0)
+    >>> np.atleast_nd(x, 2).shape
+    (1, 3)
+
+    >>> x = np.arange(12.0).reshape(4, 3)
+    >>> np.atleast_nd(x, 5).shape
+    (1, 1, 1, 4, 3)
+    >>> np.atleast_nd(x, 5).base is x.base
+    True
+
+    >>> [np.atleast_nd(x) for x in ((1, 2), [[1, 2]], [[[1, 2]]])]:
+    [array([[1, 2]]), array([[1, 2]]), array([[[1, 2]]])]
+
+    >>> np.atleast_nd((1, 2), 5, 'start').shape
+    (1, 1, 1, 1, 2)
+    >>> np.atleast_nd((1, 2), 5, 'end').shape
+    (2, 1, 1, 1, 1)
+    """
+    res = array(ary, copy=False, subok=True)
+    addons = n - res.ndim
+
+    if pos not in ('start', 'end'):
+        raise ValueError('Illegal value of pos: %s' % pos)
+
+    if addons > 0:
+        if pos == 'start' or res.ndim == 0:
```

Comment:
Is `res.ndim == 0` needed here? I'd prefer it without, so that `np.atleast_nd(a, 0)` always promotes to 0d arrays. Similarly, the line below would become `res[(newaxis,) * addons + (Ellipsis,)]`